### DISCUSSION TITLE
Rename generic to generic_ip_camera

### DISCUSSION THREAD

**Discussion by davet2001** [home-assistant/architecture#542] [General] [answered]
The IP camera integration is in 
`homeassistant/components/generic`
I propose it should be moved to 
`homeassistant/components/generic_ip_camera`

Reason:

- The code is very specific to cameras.
- It makes no sense to put other generic things in that integration any more; other generic things such as generic_thermostat are separate already.
- The name 'generic' could be more helpful (e.g. navigating code or config files).

Questions:

- Will this cause chaos?
- I've been working on a [config flow](https://github.com/home-assistant/core/pull/41019) for IP cameras.  Does moving make more sense before or after adding a config flow (I assume renaming will be a breaking change for configuration.yaml based config).

**Comment by balloob** [comment, marked as answer]
I don't think it's worth the pain

**Reply by emontnemery** [reply]
Would it be painful though, can't `generic` in `configuration.yaml` forward to `generic_ip_camera` during a transition period?

**Reply by hunterjm** [reply]
The generic IP camera is a platform component, so is defined in yaml as:

```yaml
camera:
  - platform: generic
```

This is a holdover from when the repo used to be structured differently, so unfortunately wouldn't be a super easy change.

**Reply by emontnemery** [reply]
Right, so what I had in mind was something like:
- Refactor the generic IP camera to support config flows as suggested by @davet2001
- Move the implementation from `components/generic` to `components/generic_ip_camera`
- Leave a stub implementation in `components/generic` which starts an import flow of `components/generic_ip_camera` during a deprecation period
